src/Evolution/Systems/CurvedScalarWave/Worldtube/SingletonActions/InitializeElementFaceCoordinatesMap.hpp

```diff
@@ -0,0 +1,88 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <optional>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+#include "Domain/Block.hpp"
+#include "Domain/Domain.hpp"
+#include "Domain/ElementMap.hpp"
+#include "Domain/InterfaceLogicalCoordinates.hpp"
+#include "Domain/Structure/CreateInitialMesh.hpp"
+#include "Domain/Structure/ElementId.hpp"
+#include "Domain/Structure/ExcisionSphere.hpp"
+#include "Domain/Structure/InitialElementIds.hpp"
+#include "Domain/Tags.hpp"
+#include "Evolution/DiscontinuousGalerkin/Initialization/QuadratureTag.hpp"
+#include "Evolution/Systems/CurvedScalarWave/Worldtube/Tags.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace CurvedScalarWave::Worldtube::Initialization {
+
+/*!
+ * \brief Initializes a map of the grid coordinates of all faces that abut the
+ * worldtube with corresponding ElementIds.
```

Comment:
Could you add what component this should be called on, why it needs initial extents and initial refinement, and also the associated assumptions this mutator makes? My understanding is this will get run on the Worldtube component, and assumes that no AMR happens because it explicitly uses the initial domain to compute the grid coordinates. I think that's totally fine, but should just be documented :)

Reply:
Also, could you briefly outline why you need this information on the worldtube? It's always hard to review small pieces of code without the bigger context. To _receive_ data the worldtube doesn't need to know in advance which elements are sending data. To _send_ data (or impose boundary conditions) the worldtube could just remember from which elements it received data in that timestep and respond. I mention this because with AMR this sort of element registration will become a lot more complicated, so if you can avoid element registration altogether then it will be a lot easier for you to support AMR. So if you want to support AMR, decide if you want to plan for it now or refactor later (there's overhead attached to both).

Reply:
Thank you, I added some docs. 

@nilsvu you are right, to support AMR the worldtube would need to send the coefficients of the internal solution rather than evaluate it on each face which would likely also be more efficient. However, some differences do occur as you e.g. have to figure out how to rotate the coefficients into the inertial frame (which should be easy).

I definitely want to add this later but would like to pretty much do the same thing I did for the simulations in the paper on develop first so I can have a release/commit to repeat them if necessary.

Reply:
Sounds good. Orientations are defined by the abutting_directions in the excision sphere, so available globally through the cache.